### PR TITLE
fix: crashing error when active:false and agent.registerMetrics is called

### PR DESCRIPTION
**Pull request by astorm**
Closes: #1799

This PR adds guard clauses to the `Metrics` class to prevent crashing errors when `this[registrySymbol]` is not set.  

### Checklist

- [x] Implement code
- [x] Add tests
- [x] Update documentation
- [x] Add CHANGELOG.asciidoc entry
- [x] Commit message follows [commit guidelines](https://github.com/elastic/apm-agent-nodejs/blob/master/CONTRIBUTING.md#commit-message-guidelines)

### CODE REVIEW

**Comment by astorm** [comment]
jenkins run the tests please

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -43,6 +43,7 @@ Notes:
 * Ensure the internal HTTP(S) client requests made by the APM agent to APM
   server are not themselves traced. ({issues}1168[#1168], {issues}1136[#1136])
 
+* Fix crashing error with `agent.registerMetric` and `active:false` configuration. ({issues}[#1799], {pull}[#2290])
```
Comment (trentm) [inline CHANGELOG.asciidoc:46]:
```suggestion
* Fix crashing error with `agent.registerMetric` and `active:false` configuration. ({issues}1799[#1799], {pull}2290[#2290])
```

File: test/metrics/index.test.js [hunk 2]
```diff
@@ -211,6 +211,18 @@ test('noop counter when not active', function (t) {
   t.end()
 })
 
+test('Metrics objects do not throw/crash when not started', function (t) {
```
Comment (trentm) [inline test/metrics/index.test.js:214]: nit: Could perhaps add a test between these two lines to test `agent.registerMetric(...)` before `agent.start() is called: https://github.com/elastic/apm-agent-nodejs/blob/22423a5b409b8af0a9923624e9af32e591a8159e/test/agent.test.js#L1423-L1424

Comment (astorm) [reply test/metrics/index.test.js:214]: Hmmm -- good suggestion but after looking into it my current thinking is to hold off on that -- I'm not sure that additional call brings much value to the test suite.  The `_metrics` object is mocked in that test so we're not actually hitting any of the code paths that were a problem.   Also, it's not clear what we could assert on there given that `registerMetrics` returns nothing regardless of whether the metric is created successfully or not. 

Unless you think changing `registerMetric` to return the value of the call to `this._metrics.getOrCreateGauge(name, callback, labels)` is worthwhile and not a breaking change? @trentm

Comment (trentm) [reply test/metrics/index.test.js:214]: > The _metrics object is mocked in that test so we're not actually hitting any of the code paths that were a problem. 

You had me at "this is a mock".

**Comment by astorm** [comment]
CHANGLOG links are fixed (thank you) and additional test was investigated, but ultimately discarded due to the fact we'd be testing a mock.

**Review by trentm — APPROVED** [review]

**Pull request merged**